src/main/java/org/eclipsefoundation/marketplace/resource/ListingResource.java

```diff
@@ -53,53 +62,120 @@
 	 * Endpoint for /listing/ to retrieve all listings from the database along with
 	 * the given query string parameters.
 	 * 
-	 * @param listingId string version of the listing ID. This is used as a string
-	 *                  as using an integer with bad input is handled before the
-	 *                  resource, meaning we have no input to its look and feel.
+	 * @param listingId int version of the listing ID
 	 * @return response for the browser
 	 */
 	@GET
 	public Response select() {
 		// retrieve the query parameters, and add to a modifiable map
 		QueryParams params = new QueryParams(uriInfo.getQueryParameters());
-		// TODO we should add this in a way that doesn't override existing q params
 		params.addParam(UrlParameterNames.SOLR_QUERY_STRING, "entity_type:node");
+		// retrieve the possible cached object
+		String cacheKey = getCacheKey("any", params);
+		if (LOGGER.isDebugEnabled()) {
+			LOGGER.debug("Found cache key '{}'", cacheKey);
+		}
+		Optional<Object> cachedResults = cachingService.get(cacheKey);
+
+		// if cache miss, populate cache, then return the object that is in cache
+		List<Listing> out = null;
+		if (!cachedResults.isPresent()) {
+			// retrieve the results for the parameters, and check that a listing was found
+			List<Listing> results = dao.get(SolrHelper.createQuery(params), new ListingMapper());
+			out = results;
 
-		// retrieve the results for the parameters
-		List<Listing> results = dao.get(SolrHelper.createQuery(params), ListingMapper.INSTANCE);
+			// cache the resulting listings
+			cachingService.put(cacheKey, out);
+		} else {
+			out = (List<Listing>) cachedResults.get();
+		}
```

Comment:
To avoid this whole _if cached, return; otherwise create, cache and return_ , I would define the caching service so that it leverages the `Cache::get(K, Callable<? extends V>)` method. See https://guava.dev/releases/snapshot/api/docs/com/google/common/cache/Cache.html#get-K-java.util.concurrent.Callable-

With the comment below about adding a generic type to the CachingService and moving getCacheKey to CachingService, the get method would get 3 parameters: String, QueryParams and Callable<? extends T>